crates/optimism/flashblocks/src/sequence.rs

```diff
@@ -183,4 +278,50 @@ mod tests {
 
         assert_eq!(actual_txs, expected_txs);
     }
+
+    #[test]
+    fn test_sequence_sends_flashblocks_to_subscribers() {
+        let mut sequence = FlashBlockPendingSequence::<EthereumTxEnvelope<TxEip1559>>::new();
+        let mut subscriber = sequence.subscribe_block_sequence();
+
+        for idx in 0..10 {
+            sequence
+                .insert(FlashBlock {
+                    payload_id: Default::default(),
+                    index: idx,
+                    base: None,
+                    diff: Default::default(),
+                    metadata: Default::default(),
+                })
+                .unwrap();
+        }
+
+        assert_eq!(sequence.count(), 10);
+
+        // We receive first an empty flashblock when we start building the sequence
+        let empty_flashblock = subscriber.try_recv().unwrap();
+        assert_eq!(empty_flashblock.count(), 0);
+
+        // Then we don't receive anything until we insert a new flashblock
+        let no_flashblock = subscriber.try_recv();
+        assert_eq!(no_flashblock, Err(TryRecvError::Empty));
+
+        // Let's insert a new flashblock with index 0
+        sequence
+            .insert(FlashBlock {
+                payload_id: Default::default(),
+                index: 0,
+                base: None,
```

Comment:
This is not really an issue of this PR, but rather #18272

If a complete flash block sequence means that it can be used to build a block, then being able to get it without the `base` does not make sense.

I think the `base` needs to be an invariant which, if not met, forbids the construction of the complete sequence. Otherwise, it would be vulnerable to panics.

Reply:
Agree, check the FlashBlockCompleteSequence::new and TryFrom impl from #18272, which try to ensure that.